src/core/elf_file.cpp

```diff
@@ -162,8 +162,8 @@ namespace linuxdeploy {
 
             ElfFile::ElfFile(const boost::filesystem::path& path) {
                 // check if file exists
-                if (!bf::exists(path))
-                    throw ElfFileParseError("No such file or directory: " + path.string());
+                if (!bf::exists(path) || !bf::is_regular_file(path))
```

Comment:
I think `exists` is redundant, isn't it? I haven't checked the docs.

We should switch to C++17...

Reply:
Indeed, `exists` is redundant here. Please fix.